### ISSUE TITLE
Add tabindex to highlight template func markup

### ISSUE THREAD

**Issue by gpanders** [gohugoio/hugo#7194]
Code blocks generated through the `{{ highlight }}` shortcode or through the use of Markdown code fences should have the `tabindex="0"` attribute, allowing them to be tab focusable. This is good for accessibility; see [here](https://dequeuniversity.com/rules/axe/3.5/scrollable-region-focusable).

I've searched through the docs and I cannot find any way to set this attribute myself in my own site, since the `highlight` shortcode is (apparently) built-in to Hugo, so I don't think there is any way to do this outside of a change to Hugo itself.

**Comment by bep** [comment]
Never mind that comment, I see that the shortcode is just delegating to the internal template func.

**Comment by faraixyz** [comment]
The internal function wraps the code in a div tag.

https://github.com/gohugoio/hugo/blob/f0266e2ef3487bc57dd05402002fc816e3b40195/markup/highlight/highlight.go#L88-L92

Would it be possible to move this to the short code?

https://github.com/gohugoio/hugo/blob/64789fb5dcf8326f14f13d69a2576ae3aa2bbbaa/tpl/tplimpl/embedded/templates/shortcodes/highlight.html#L1

That way users have more flexibility in wrapping the generated `pre` block.

**Comment by faraixyz** [comment]
Hey again, I was thinking of this again since https://github.com/gohugoio/hugo/pull/8280 with attribute lists for code fences. Would it be possible to have a way to specify default attributes for code fences?

**Comment by rhymes** [comment]
@sephore I have opened a PR to fix it here https://github.com/gohugoio/hugo/pull/8568 - I used Firexo and generated a demo Hugo site with highlighting and I wasn't able to focus on `<pre>`, you need that for accessibility purposes, see the documentation in the PR
